adopterAPI.json

```diff
@@ -0,0 +1,457 @@
+{
+  "openapi": "3.0.1",
+  "info": {
+    "title": "AdopterAPI",
+    "version": "v1"
+  },
+  "paths": {
+    "/adopter": {
+      "get": {
+        "tags": [
+          "AdopterEndpoints"
+        ],
+        "summary": "Get all adopters. Requires admin role.",
+        "responses": {
+          "200": {
+            "description": "Success"
+          },
+          "401": {
+            "description": "Client error"
+          }
+        }
+      },
+      "post": {
+        "tags": [
+          "AdopterEndpoints"
+        ],
+        "summary": "Posts new adopter.",
```

Comment:
I am suggesting to specify, who is able to post new adopter. In other endpoints authorization is mentioned in summaries.

Reply:
It's a bit more tricky. This post is called before we have any role assigned, so it is open for any role (including non-assigned) because it is used in registration process. I can't really see a better way

Reply:
We could specify that explicitly for clarity. But IMO it's ok as it is now.

Reply:
Ooor we could decide that this endpoint _requires_ `unassigned` role. It doesn't make sense for i.e. a shelter to `POST` an adopter.

Reply:
Requiring `unassigend` role don't make much of a difference in my opinion

Reply:
In my opinion, we should somehow secure this endpoint, because currently anyone can create a million users in our database

Reply:
> because currently anyone can create a million users in our database

Not really, it still does require a valid token. 

Unassigned role may be a solution, if everyone agrees on that I'll change it